contrib/generic_proxy/filters/network/source/codecs/dubbo/config.cc

```diff
@@ -0,0 +1,175 @@
+#include "contrib/generic_proxy/filters/network/source/codecs/dubbo/config.h"
+
+#include <memory>
+
+#include "envoy/registry/registry.h"
+
+#include "source/extensions/common/dubbo/message_impl.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace GenericProxy {
+namespace Codec {
+namespace Dubbo {
+
+namespace {
+
+constexpr absl::string_view VERSION_KEY = "version";
+
+#define ENUM_TO_STRING_VIEW(X)                                                                     \
+  case Common::Dubbo::ResponseStatus::X:                                                           \
+    static constexpr absl::string_view X##_VIEW = #X;                                              \
+    return X##_VIEW;
+
+absl::string_view responseStatusToStringView(Common::Dubbo::ResponseStatus status) {
+  switch (status) {
+    ENUM_TO_STRING_VIEW(Ok);
+    ENUM_TO_STRING_VIEW(ClientTimeout);
+    ENUM_TO_STRING_VIEW(ServerTimeout);
+    ENUM_TO_STRING_VIEW(BadRequest);
+    ENUM_TO_STRING_VIEW(BadResponse);
+    ENUM_TO_STRING_VIEW(ServiceNotFound);
+    ENUM_TO_STRING_VIEW(ServiceError);
+    ENUM_TO_STRING_VIEW(ServerError);
+    ENUM_TO_STRING_VIEW(ClientError);
+    ENUM_TO_STRING_VIEW(ServerThreadpoolExhaustedError);
+  }
+}
+
+Common::Dubbo::ResponseStatus genericStatusToStatus(StatusCode code) {
+  switch (code) {
+  case StatusCode::kOk:
+    return Common::Dubbo::ResponseStatus::Ok;
+  case StatusCode::kInvalidArgument:
+    return Common::Dubbo::ResponseStatus::BadRequest;
+  default:
+    return Common::Dubbo::ResponseStatus::ServerError;
+  }
+}
+
+StatusCode statusToGenericStatus(Common::Dubbo::ResponseStatus status) {
```

Comment:
These two functions don't seem to commute. Should they somehow map statuses one-to-one?

Reply:
there is no way to map Status (alias of absl::Status) and ResponseStatus (defined by dubbo protocol) one-to-one.
this is an expected flaw.